visualization/app/codeCharta/ui/attributeSideBar/attributeSideBar.component.ts

```diff
@@ -108,9 +117,27 @@ export class AttributeSideBarController
 	private updateSortedMetricKeysWithoutPrimaryMetrics() {
 		if (this._viewModel.node) {
 			const metricValues = new Set(Object.values(this._viewModel.primaryMetricKeys.node))
-			this._viewModel.secondaryMetricKeys = Object.keys(this._viewModel.node.attributes)
+
+			const secondaryMetricKeys = Object.keys(this._viewModel.node.attributes)
 				.filter(key => !metricValues.has(key))
 				.sort()
+
+			const secondaryMetrics = []
+
+			for (const metricKey of secondaryMetricKeys) {
+				// Temporary solution to remove unary as viewable metric
+				if (metricKey === "unary") {
+					continue
+				}
+
+				const metricType =
+					metricKey in this._viewModel.node.edgeAttributes
+						? this.edgeMetricDataService.getAttributeTypeByMetric(metricKey)
+						: this.nodeMetricDataService.getAttributeTypeByMetric(metricKey)
+				secondaryMetrics.push({ name: metricKey, type: metricType } as SecondaryMetric)
+			}
+
+			this._viewModel.secondaryMetrics = secondaryMetrics
```

Comment:
We should move this code to a redux property, I think
Store it in the state and then we can access it and react to changes